File: Lab 3/shape/src/main/java/Shape.java
public abstract class Shape {
    protected double dim;

    public Shape() {
    }

    public Shape(double dim) {
        this.dim = dim;
    }

    public abstract double calculateArea();

    public double getDim() {
        return dim;
    }

    public void setDim(double dim) {
        this.dim = dim;
    }
}

class Rectangle extends Shape{
    private double width;

    public Rectangle(){
    }



    public Rectangle(double dim,double width){
        super(dim);
        this.width = width;
    }

    @Override
    public double calculateArea() {
        return dim * width;
    }

}
class Triangle extends Shape{
    private double base;

    public Triangle() {
    }

    public Triangle(double dim,double base) {
        super(dim);
        this.base = base;
    }

    @Override
    public double calculateArea() {
        return (0.5 * base) * dim;
    }
}
class Circle extends Shape{
    private final double PI = 3.14;
    public Circle() {
    }

    public Circle(double dim) {
        super(dim);
    }

    @Override
    public double calculateArea() {
        return PI * (dim * dim);
    }
}

File: Kotlin/Lab_2/src/main/kotlin/Shape.kt
abstract class Shape(var dim : Double) {

    abstract fun calcArea() : Double

}

class Rectangle(dim : Double,var height: Double) : Shape(dim){

    override fun calcArea(): Double {
        return dim * height
    }
}

class Triangle(dim : Double,var height: Double) : Shape(dim){

    override fun calcArea(): Double {
        return (0.5 * dim) * height
    }

}

class Circle(dim : Double) : Shape(dim){

    override fun calcArea(): Double {
        return 3.14 * (dim * dim)
    }

}

class Picture(){
    fun sumAreas(s1 : Shape,s2 : Shape, s3 : Shape) : Double{
        return s1.calcArea() + s2.calcArea() + s3.calcArea()
    }
}
fun main(){
    val picture = Picture()
    val rect = Rectangle(3.0,2.0) // 6
    val circle = Circle(4.0) //
    val triangle = Triangle(2.0,4.0) // 4

    println("Sum Areas = ${picture.sumAreas(rect,circle,triangle)}")


}